src/components/text-editor/prosemirror-adapter/plugins/trigger-popup-plugin.ts

```diff
@@ -0,0 +1,125 @@
+import { Plugin, PluginKey, EditorState } from 'prosemirror-state';
+import { ReplaceStep, ReplaceAroundStep } from 'prosemirror-transform';
+import { EditorView } from 'prosemirror-view';
+
+export const triggerPluginKey = new PluginKey('triggerPlugin');
+
+type Trigger = '@' | '#';
+
+const triggers: Trigger[] = ['@', '#'];
+
+const isTrigger = (key: string): key is Trigger => {
+    return triggers.includes(key as Trigger);
+};
+
+const shouldTrigger = (state: EditorState): boolean => {
```

Comment:
these short command trigger should also work if the previous character is a line-break. So when `@` or `#` are the first letters of a new paragraph.

Reply:
seems they already do that ;)